src/plugins/discover/public/application/apps/main/components/chart/discover_chart.tsx

```diff
@@ -38,6 +41,8 @@ export function DiscoverChart({
   state: AppState;
   stateContainer: GetStateReturn;
   timefield?: string;
+  discoverViewMode: string;
+  setDiscoverViewMode: (discoverViewMode: string) => void;
```

Comment:
:art: We should not type this generic as `string`, but a union of the actual accepted values, or even better (as suggested below) make this an `enum` instead. This applies for multiple places around this PR, that currently use `string` for passing on this value.

Reply:
Updated here https://github.com/elastic/kibana/pull/107184/commits/f12711a802b6208952ba700785aa83a6cdddb615